### PR TITLE
Add packaging for debian based linux distributions

### PR DESCRIPTION
**Pull request by allonsy**
Hey friends!
Been a while!
I finally got around to finishing the debian packaging work.
I added a package directory with documentation and scripts on auto-generating the deb file.
We can attach the generated deb files to the releases tab in github for users to download and then they can just run `dpkg` on the deb file or use their favorite gui package manager.
Let me know what you think!

### CODE REVIEW

File: package/Packaging.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Git-Town Packaging and Release Process
+
+Instructions on how to update all the necessary packages once a new release is out
+
+## Linux
+The current linux (all distros) maintainer is [allonsy](https://github.com/allonsy). If an update occurs, please bug him
+### Arch Linux
```
Comment (kevgo) [inline package/Packaging.md]: Add empty line above this.

File: package/Packaging.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+The package will need to be flagged out of date [here](https://aur.archlinux.org/packages/git-town/)
+This will tell [allonsy](https://github.com/allonsy) to update the package (most likely minimal intervention necessary). There is a button on the right hand side menu to flag the package.
+
+
+### Debian Based Distributions
+The package will need to be repackaged into a deb tarball.
+There are two options: rebuild it from your own debian based machine or build it on the included docker image (recommended)
+
+For Either process you will need to update the version number for the package located in the following configuration files:
```
Comment (kevgo) [inline package/Packaging.md]: Lowercase `Either`

File: package/Packaging.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Git-Town Packaging and Release Process
```
Comment (kevgo) [inline package/Packaging.md]: Can you lowercase the filename?

Comment (kevgo) [inline package/Packaging.md]: Can you add that these instructions are for Linux?

**Comment by kevgo** [comment]
Hey Alec, super great to hear from you! Thanks a ton for doing this! Looks good to me! I have given you contributor access, so you can `git ship` this PR!

**Comment by kevgo** [comment]
Or, for simplicity sake, I have merged it in. From now on you can work off our repo directly!

**Comment by allonsy** [comment]
Thanks! Great to hear from you too!
I agree with all of your comments. Do you want me to implement those changes in a separate PR?

**Comment by allonsy** [comment]
I've added a backwards release for v0.10.0 (since my scripts will only work for future releases where a package directory exists) to the releases page on github until I can sort out the mess that is debian software source servers

**Comment by kevgo** [comment]
Ah, I forgot there are comments on this PR. Yes, please do them in a separate one, haha.

**Pull request merged**